packages/contracts-bedrock/foundry.toml

```diff
@@ -68,8 +70,14 @@ optimizer = false
 #                         PROFILE: KONTROL                     #
 ################################################################
 
-[profile.kontrol]
+[profile.kdeploy]
 src = 'src/L1'
-out = 'kout'
+out = 'kout-deployment'
 test = 'test/kontrol'
 script = 'scripts-kontrol'
+
+[profile.kprove]
+src = 'test/kontrol/proofs'
+out = 'kout-proofs'
+test = 'test/kontrol/proofs'
+script = 'scripts-kontrol'
```

Comment:
Can we add some comments explaining these profiles and how they relate to each other? I think I understand from a high level: first run with `kdeploy` profile to get the state diff, then run with `krpove` profile to run the proofs. 

Some specific questions that I'd want the comments to answer:
- Presumably kprove loads artifacts from the kdeploy directory?
- The `src = 'test/...` is confusing and might seem like a mistake without comments explaining it
- Why we have a `scripts-kontrol` here even though that folder doesn't exist

Reply:
Good point. I'll answer them here first in case we have to iterate.
- `kprove` uses no artifacts from the  `kout-deployment` directory. The `kdeploy` profile will only use the necessary files to generate the deployment summary and save it under `kontrol/proofs/utils`. The `kprove` profile only uses the files under the `kontrol/proofs` dir, which have no dependencies on any files outside of that folder
- I'll add, "We only want one directory in scope"
- I'll add, "We want the script folder to be empty"
Does this seem fine?

Reply:
Based on your replies, here's my draft of proposed comments, you can see I still have a question in there. Feedback welcome!

```toml
# This profile is used to generate the deployment summary solidity contract, which is used by the
# `kprove` profile to load the post-setUp state directly into kontrol. We don't need the output artifacts
# from this step, so we save them to the `kout-deployment` directory, which is not used elsewhere.
# We also point the script path to the `scripts-kontrol` directory, which does not exist, to avoid
# compiling scripts we don't need which reduces execution time.
[profile.kdeploy]
src = 'src/L1'
out = 'kout'
out = 'kout-deployment'
test = 'test/kontrol'
script = 'scripts-kontrol'

# This profile is used after running `<command>` with the `kdeploy` profile, and is responsible for
# actually executing the proofs. The `src` directory points to a test folder because <?? I still
# don't understand this part, how does it not need to compile src files?>. We similarly point the
# script path to a non-existent directory for the same reason as above.
[profile.kprove]
src = 'test/kontrol/proofs'
out = 'kout-proofs'
test = 'test/kontrol/proofs'
script = 'scripts-kontrol'
```

Reply:
>The `src` directory points to a test folder because

We only want to compile what is in the `test/kontrol/proofs` folder since it contains all the deployed code and the proofs. That is, the `test/kontrol/proofs` is isolated entirely from the rest of the repo and contains all the necessary information to verify. That's why we don't want to compile anything outside of it

Reply:
Adding something that might not be clear: the compilation of the necessary `src/L1` files is done with the `kdeploy` profile, and the results are saved into `test/kontrol/proofs/utils`. So, when running the `kprove` profile, the compiled `src/L1` files are in `test/kontrol/proofs/utils/DeploymentSummaryCode.sol`

Reply:
Got it, that all makes sense! Let's instead put all this info in the README—since it explains the overall setup—and add a comment to `foundry.toml` that says something like "See test/kontrol/README.md for an explanation of how the profiles are configured"